#404 Historical Cases chart on facility page

daschi opened the pull request:
## Description of the change

I'm opening this PR as a draft to keep everyone updated on the progress and what is left to do. If anyone is interested in working on this too I think some of these tasks could be worked on separately! We could use this PR to track progress, and if someone wants to work on one of the todos, I was thinking you could just edit this and put your name next to it? Totally open to other ideas, let me know what you think. 

What I'm working on next: 
- [x] Use the sum of cases for incarcerated and staff counts
- [x] Add ability to click to `next 15 days` and `previous 15 days`
- [x] (Daniela)Add text in tooltip that says "Click to update"
- [x] (Daniela) Add an onClick to the bars to open the Add Cases modal with auto-populated date from the clicked bar and auto-populated cases data if it exists
- [x] (Daniela) Format the font styles in the tooltip, pull the tooltip out into a generic component that can be shared


**TODO:** 

- [x] [Soumya #459 ]Change the bar's color to a darker color on hover to signify it as a click target
- [x] [Soumya #459 ] Add the chart header with the Cases title and a button that opens the Add Cases modal with yesterday's date pre-selected (or what the current default date is)
- [x] [Daniela] Format the style of the y and x axis font & color
- [x] [Daniela] Chart width becomes too wide when screen size becomes smaller

**TBD**
- [ ] **P2/might not implement**: Display month's name under the x-axis at certain points (?)
- [ ]  **TBD**: If case count is 0 or less than 50% of the y-axis, display the count above the bar
- [ ] **TBD w/Serena first** Display empty bar chart when adding a new facility and allow user to add historical data in the add cases modal for a new facility


Some possible blockers:

- ~I still need to investigate whether it'll be possible to position the tooltip on hover so that a user can click on a link inside it~
- ~I had some trouble figuring out how to display the x axis with the two rows of labels (i.e. the date and the month underneath the 1), so for now I formatted the x-axis label as `MM/dd`~
- **Nice-to-have:** Currently the tooltip appears at the top of the bar (or at the bottom if there is no data), I'm not sure if we want to change this so that it sometimes appears in the middle of the bar?

**Note:** @sychang and I chatted about removing the red bar that is shown in the mocks, and showing no bar for now when there are 0 cases. 

Here's where it is so far visually (see linked issue for the mock):
![image](https://user-images.githubusercontent.com/5402804/81998388-38329d00-9607-11ea-8172-9a9ee4802322.png)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #404 

## Checklists

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

sychang commented:
For the blockers:
I still need to investigate whether it'll be possible to position the tooltip on hover so that a user can click on a link inside it 
--> Let's do away with the link inside the tooltip and make it say "click to update" with no link instead for simplification

I had some trouble figuring out how to display the x axis with the two rows of labels (i.e. the date and the month underneath the 1), so for now I formatted the x-axis label as MM/dd 
--> MM/dd is totally fine, let's do away with the two rows of labels for simplification

Currently the tooltip appears at the top of the bar (or at the bottom if there is no data), I'm not sure if we want to change this so that it sometimes appears in the middle of the bar?
--> I would say this is nice-to-have but definitely not a launch-blocker

Now that we are no longer using the red bars to show missing data, I would change this one slightly: "P2: If case count is 0 or less than 50% of the y-axis, display the count above the bar" 
--> The minimum would be visually distinguishing "0 cases" from "missing data" (which would both currently appear to be empty bars). Whoever ends up taking that component, let's chat about through a few options that are easiest. TODO: I will come back and update this comment with some options and what we decide

daschi commented:
@jessex I believe this PR along with @SoumyaMahadevan's PR #459 are ready for review. Let us know if it would be easier to review them together in one PR

jessex commented:
Thanks, @daschi and @SoumyaMahadevan!

My plan is to test on Soumya's PR, review Daniela's PR, then review Soumya's PR. Then assuming all is good, the easiest path is probably to merge Soumya's PR into Daniela's, and then merge Daniela's into master.

rocktacular commented:
ok got through it all and think I understand what's happening. Lots of new functionality and seems fine to me. I haven't looked at Soumya's PR, but I agree with Josh that pulling hers into this one before going to master is probably easiest.

I see 1 linting issue you may want to address.

jessex commented on src/hooks/useAddCasesInputs.tsx:
```diff
@@ -0,0 +1,141 @@
+    defaultObservationDate,
+  );
+
+  // maybe export this so that we don't define it twice (it is also in the AddCasesModalContent component)
```
Would be good, but I also don't mind filing a tech debt issue to loop back around and handle this, given how much effort has already gone into this feature and its overall importance!

daschi replied:
Okay sounds good, I'll remove the note for now.

jessex commented on src/hooks/useAddCasesInputs.tsx:
```diff
@@ -0,0 +1,141 @@
+    }
+  }, [observedAt]);
+
+  // TODO: Handle add cases modal for new facility?
```
This todo is irrelevant, right? The modal isn't shown when you're creating a brand new facility.

daschi replied:
Yes I can remove this! thanks

daschi commented:
@jessex and @rocktacular thanks for the reviews! I've updated it so there's more margin on the right side of the x-axis and added the toast back in (oops). I think this is ready for another look, let me know if there's any other issues.
